### PR TITLE
Fix recurrent dropout for GRU.

### PR DESCRIPTION
**Pull request by hertschuh**
The simplified implementation, which used the same recurrent dropout masks for all the previous states didn't work and caused the training to not converge with large enough recurrent dropout values.

This new implementation is now the same as Keras 2. Note that recurrent dropout requires "implementation 1" to be turned on.

Fixes https://github.com/keras-team/keras/issues/20276

### CODE REVIEW

**Review by fchollet — APPROVED** [review]
LGTM, thanks for the fix!
